lib/KeyboardShortcuts.tsx

```diff
@@ -1,31 +1,109 @@
-'use client';
-
 import React from 'react';
 import { Track } from 'livekit-client';
 import { useTrackToggle } from '@livekit/components-react';
+import { useSettingsState } from './SettingsContext';
+import { KeyCommand } from './types';
 
 export function KeyboardShortcuts() {
-  const { toggle: toggleMic } = useTrackToggle({ source: Track.Source.Microphone });
+  const { state } = useSettingsState() ?? {};
+  const { toggle: toggleMic, enabled: micEnabled } = useTrackToggle({
+    source: Track.Source.Microphone,
+  });
   const { toggle: toggleCamera } = useTrackToggle({ source: Track.Source.Camera });
+  const [pttHeld, setPttHeld] = React.useState(false);
 
   React.useEffect(() => {
-    function handleShortcut(event: KeyboardEvent) {
-      // Toggle microphone: Cmd/Ctrl-Shift-A
-      if (toggleMic && event.key === 'A' && (event.ctrlKey || event.metaKey)) {
-        event.preventDefault();
-        toggleMic();
-      }
-
-      // Toggle camera: Cmd/Ctrl-Shift-V
-      if (event.key === 'V' && (event.ctrlKey || event.metaKey)) {
-        event.preventDefault();
-        toggleCamera();
-      }
-    }
-
-    window.addEventListener('keydown', handleShortcut);
-    return () => window.removeEventListener('keydown', handleShortcut);
-  }, [toggleMic, toggleCamera]);
+    const handlers = Object.entries(state.keybindings)
+      .flatMap(([command, bind]) => {
+        switch (command) {
+          case KeyCommand.PTT:
+            if (!state.enablePTT || !Array.isArray(bind)) return [];
+
+            const [enable, disable] = bind;
+            const t = getEventTarget(enable.target);
+            if (!t) return null;
+
+            const on = (event: KeyboardEvent) => {
+              if (enable.discriminator(event)) {
+                event.preventDefault();
+                if (!micEnabled) {
```

Comment:
I think this will race. The `useTrackToggle` hook also returns `pending` state which is between state change and its state change trigger. 

Generally explicitly muting/unmuting might be a better fit for PTT than toggling ?

Reply:
Done. And now checking the pending state before firing another `toggle` action in the other cases.

Reply:
Anecdotally, I have spammed both PTT and toggle as fast as I can and don't see any unexpected behavior.